ASCIT/donut
# Voting

### Summary
- Support for surveys/elections:
  - Can create and edit surveys, take them, and view the responses once they close
  - Currently supported question types: dropdown (select one from a list), checkboxes (select a subset of a list), elected position (allows ordering a list, submitting write-ins, and NO; results of ranked pairs are shown), short text (input), and long text (textarea)
  - Surveys can be restricted to members of a specific group
  - A survey can be configured not to appear in the list of active surveys so that a link is required to take it
  - Can also require survey-takers to authenticate with their UID and birthday
  - Survey results are available to the creator once the survey closes, but must be explicitly released by the creator so others can see them
- Refactored the directory search name results display so it is easy to add it to other pages
- Also figured out why tests were taking 2 minutes to load and fixed it

### Test Plan
- Try creating some surveys, editing their questions and settings
- Make sure questions editor has all the features we might want to provide
- Have a bunch of users take a test survey so the results can be verified
